homeassistant/components/nordpool/sensor.py

```diff
@@ -34,9 +34,10 @@ def validate_prices(
     index: int,
 ) -> float | None:
     """Validate and return."""
-    if result := func(entity)[area][index]:
-        return result / 1000
-    return None
+    result = func(entity)[area][index]
+    if result is None:
+        return None
+    return result / 1000 if result != 0 else 0
```

Comment:
Let's keep the original walrus and remove the in-line if/else construct.

```suggestion    
    if (result := func(entity)[area][index]) is None:
        return None
    if result == 0:
        return 0
    return result / 1000
```

Reply:
Yes. Now more clean